pontoon/base/static/js/locale_project.js

```diff
@@ -0,0 +1,65 @@
+$(function() {
+
+  // Search entire project with Enter (and remove default search handler)
+  $('.part .menu .search-project input[type=search]').off('keyup.search').keydown(function (e) {
+    if (e.which === 13) {
+      e.stopPropagation();
+
+      var q = $(this).val(),
+          action = '';
+
+      if (q.length > 0) {
+        if ($('.locale .selector .language').length) {
+          var locale = $('.locale .selector .language').attr('class').split(' ')[1],
+              project = $('.project .selector .title').data('slug'),
+              action = '/' + locale + '/' + project + '/';
+        }
+
+        var options = '';
+        $('.search-project .options li').each(function() {
```

Comment:
The `<form>` comment above would also help here, as the search options would just be checkboxes included in the form's data.